### PR TITLE
Supply `protoc` plugin via JAR 

### PR DESCRIPTION
**Pull request by dmdashenkov**
The latest version of Protobuf Gradle plugin allows using a runnable JAR as a `protoc` plugin instead of an executable file. This means less work for us.

This PR also updates third-party dependencies, including ErrorProne. New ErrorProne warnings are addressed.

### CODE REVIEW

File: base/src/main/java/io/spine/type/UnknownTypeException.java [hunk 1]
```diff
@@ -42,7 +39,7 @@ public UnknownTypeException(String typeName) {
     }
 
     private static String makeMsg(String typeName) {
-        return format(ERR_MSG_NO_JAVA_CLASS_FOUND, typeName);
+        return format("No Java class found for the Protobuf message of type: %s", typeName);
```
Comment (alexander-yevsyukov) [inline base/src/main/java/io/spine/type/UnknownTypeException.java:42]: Please backtick the type, and put period at the end.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov, @armiol, PTAL.

File: tools/proto-js-plugin/src/main/java/io/spine/js/generate/field/parser/PrimitiveFieldParser.java [hunk 2]
```diff
@@ -37,9 +37,11 @@
 @SuppressWarnings("BadImport") // refer to `FieldDescriptor.Type` as `Type` in this class.
```
Comment (alexander-yevsyukov) [inline tools/proto-js-plugin/src/main/java/io/spine/js/generate/field/parser/PrimitiveFieldParser.java:37]: Do we need this suppression, assuming we have them in-place below?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with one minor question.

File: tools/plugin-base/src/main/java/io/spine/tools/groovy/GStrings.java [hunk 3]
```diff
@@ -24,7 +24,11 @@
 
 /**
  * A factory of Groovy strings.
+ *
+ * @deprecated Since the latest changes in the Protobuf Gradle plugin, usage Groovy strings in Java
```
Comment (armiol) [inline tools/plugin-base/src/main/java/io/spine/tools/groovy/GStrings.java:28]: I think it's `of Groovy`.

File: tools/protoc-plugin/build.gradle.kts [hunk 4]
```diff
@@ -54,4 +54,7 @@ tasks.jar {
             else -> zipTree(it)
         }
     })
+    // We should provide a classifier or else Protobuf Gradle plugin will substitute an OS-specific
```
Comment (armiol) [inline tools/protoc-plugin/build.gradle.kts:57]: Sorry to bother you again on the same matter, but I think it's `will substitute it with an OS-specific one.`

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM in general. Please see my comments though.

**Pull request merged**